src/Utils/ArrayUtils.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * ArrayUtils class file.
+ *
+ * @package WooCommerce\Utils
+ */
+
+namespace Automattic\WooCommerce\Utils;
+
+use Automattic\WooCommerce\Utils\StringUtils;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Miscellaneous array related utility methods.
+ *
+ * @package WooCommerce\Utils
+ */
+class ArrayUtils {
+
+	/**
+	 * Get a value from an nested array by specifying the entire key hierarchy with '::' as separator.
+	 *
+	 * E.g. for [ 'foo' => [ 'bar' => [ 'fizz' => 'buzz' ] ] ] the value for key 'foo::bar::fizz' would be 'buzz'.
+	 *
+	 * @param array  $array The array to get the value from.
+	 * @param string $key The complete key hierarchy, using '::' as separator.
+	 * @param mixed  $default The value to return if the key doesn't exist in the array.
+	 *
+	 * @return mixed The retrieved value, or the supplied default value.
+	 * @throws \Exception $array is not an array.
+	 */
+	public static function get_nested_value( $array, $key, $default = null ) {
+		if ( ! is_array( $array ) ) {
+			throw new \Exception( StringUtils::get_class_short_name( __CLASS__ ) . '::' . __METHOD__ . ': $array must be an array.' );
+		}
+
+		$key_stack = explode( '::', $key );
```

Comment:
Dot notation seems to be a more common pattern for these kinds of utilities in the PHP ecosystem, what do you think?

Reply:
The thing is that this is being used by the replacement of the `wc_get_theme_support` method, and we need to keep this separator for compatibility. But I could make the separator an optional parameter of the method, defaulting to `.`, and use `::` when invoked from the theme support class.

Reply:
It looks like `::` is a pretty common delimiter in the WordPress ecosystem so let's just leave it as it is I think!